packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -966,35 +1010,78 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
 
   final Map<int, Widget> _childWidgets = HashMap<int, Widget>();
   final SplayTreeMap<int, Element> _childElements = SplayTreeMap<int, Element>();
-  RenderBox _currentBeforeChild;
 
   @override
   void performRebuild() {
     _childWidgets.clear(); // Reset the cache, as described above.
     super.performRebuild();
-    _currentBeforeChild = null;
     assert(_currentlyUpdatingChildIndex == null);
     try {
+      final Set<int> affectedIndexes = Set<int>.from(_childElements.keys);
+      final Map<int, Element> newKeepAliveChildren = <int, Element>{};
```

Comment:
Do you need the set and the map? Isn't the set represented by Map.keys? Also, if you use a SplayTreeMap the keys would be automatically ordered?